trace-obfuscation/src/credit_cards.rs

```diff
@@ -0,0 +1,419 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+/// is_card_number checks if b could be a credit card number by checking the digit count and IIN prefix.
+/// If validateLuhn is true, the Luhn checksum is also applied to potential candidates.
+/// Note: This code is based on the code from datadog-agent/pkg/obfuscate/credit_cards.go
+pub fn is_card_number(s: &str, validate_luhn: bool) -> bool {
+    if s.len() < 12 {
+        // fast path
+        return false;
+    }
+
+    let mut num_s = Vec::with_capacity(s.len());
+    for c in s.chars() {
+        // Only valid characters are 0-9, space (" ") and dash("-")
+        match c {
+            ' ' | '-' => continue,
+            _ => (),
+        };
+        if !c.is_numeric() {
```

Comment:
Is this supposed to be ascii numeric? Rust's chars are not the same as C chars, FYI.

Reply:
These would also be good test cases to add :)

Reply:
Oooo that's good to know, yeah I think this should be `is_ascii_digit`  I'll fix this